src/SampleAggregator.cpp

```diff
@@ -419,4 +458,25 @@ namespace geopm
     {
         return sample_epoch_helper(signal_idx, true);
     }
+
+    double SampleAggregatorImp::sample_period_last(int signal_idx)
+    {
+        double result = NAN;
+        if (m_period_duration != 0.0) {
+            auto sum_it = m_sum_signal.find(signal_idx);
+            if (sum_it != m_sum_signal.end()) {
+                result = sum_it->second.period_accum->interval_total();
+            }
+            else {
+                auto avg_it = m_avg_signal.find(signal_idx);
```

Comment:
no test for this branch of the code

Reply:
This is a general problem for this unit test, not particular to this feature.  I opened an issue here:

https://github.com/geopm/geopm/issues/1590